test/run_test.py

```diff
@@ -1028,6 +1028,49 @@ def export_S3_test_times(test_times_filename: str, test_times: Dict[str, float])
         json.dump(job_times_json, file, indent='    ', separators=(',', ': '))
         file.write('\n')
 
+
+def query_changed_test_files() -> List[str]:
+    cmd = ["git", "diff", "--name-only", "origin/master", "HEAD"]
```

Comment:
not sure if this is related but I recently ran into huge list of tests being reordered because it is comparing against HEAD instead of viable strict. 
should we consider 
1. changing `"HEAD"` to `"$(git merge-base HEAD origin/master)"`
2. not enabling this query for local development?